examples/sequential/requirements-dev.txt

```diff
@@ -0,0 +1,3 @@
+pytest>=3.5.0
+pip>=21.3
+maturin>=0.12,<0.13
```

Comment:
Seems redundant to the Nox session and inconsistent with pyproject.toml? Maybe drop the requirements file outright and rely solely on the Nox session?

Reply:
Probably there's a broader tidy up we should consider for all of these examples, I suspect this was copypasta too.

Reply:
Resolved separately in #3536, I'll rebase this.